aexpect/remote.py

```diff
@@ -75,6 +75,20 @@ def __str__(self):
         return "%s    (output: %r)" % (self.msg, self.output)
 
 
+class TransferBadClientError(TransferError):
```

Comment:
Looking at this hierarchy, the `msg` and `output` doesn't really scale here. How about introducing simple `RemoteError(Exception):  pass` and inherit from that? This Exception would be then used also in https://github.com/avocado-framework/aexpect/pull/55/commits/3a25533c4f7a4f21d0f9464e8f8ee20b6b8d993c#r291033819 as no output is available there...

Reply:
Good point, corrected with the next push. Please take another look before I resolve this issue so that we make sure the exception hierarchy matches your expectations fully.